src/server/game/Entities/Totem/Totem.cpp

```diff
@@ -157,9 +157,13 @@ void Totem::UnSummon(uint32 msTime)
 
 bool Totem::IsImmunedToSpellEffect(SpellInfo const* spellInfo, uint32 index, WorldObject const* caster) const
 {
-    /// @todo possibly all negative auras immune?
-    if (GetEntry() == 5925)
-        return false;
+    // immune to all positive spells, except of stoneclaw totem absorb and sentry totem bind sight
+	// totems positive spells have unit_caster target
+    if (spellInfo->Effects[index].Effect != SPELL_EFFECT_DUMMY && 
+        spellInfo->Effects[index].Effect != SPELL_EFFECT_SCRIPT_EFFECT && 
+        spellInfo->IsPositive() && spellInfo->Effects[index].TargetA.GetTarget() != TARGET_UNIT_CASTER &&
+        spellInfo->Effects[index].TargetA.GetCheckType() != TARGET_CHECK_ENTRY && spellInfo->Id != 55277 && spellInfo->Id != 6277)
```

Comment:
avoid using magic numbers, like 55277, 6277

Reply:
> avoid using magic numbers, like 55277, 6277

Do I declare them in Totem.h? I'm just learning)

Reply:
Done, check pls.